### ISSUE TITLE
Force model to use FP32

### ISSUE THREAD

**Issue by hukkelas** [NVIDIA/apex#178]
I'm currently implementing a generative adversarial network, and I have a exponential running average of the generator. The running average copies the weight of the trained model for each step, where we keep about 99.9% of the current running average of the model and 0.1% of the trained model. 

This is not working with AMP enabled and I believe the running average requires fp32 precision to work well. I would like to keep the trained models, the generator/discriminator to traing with AMP, while keeping the running average updates with fp32 precision. Is it possible to force the running average to be fp32? 

FYI, this is the update function I'm running each training step
```python
    @amp.float_function
    def update_running_average_generator(self):
        for avg_parameter, current_parameter in zip(
                self.running_average_generator.parameters(),
                self.generator.parameters()):
            avg_parameter.data = 0.999*avg_parameter + 0.001 * current_parameter
```

**Comment by mcarilli** [comment]
It might not be a problem with Amp in particular, but rather, a tricky interaction of dynamic loss scaling with GANs.  I'm working on a comprehensive GAN example using the new Amp API, which people are strongly encouraged to use:
https://nvidia.github.io/apex/amp.html
https://nvidia.github.io/apex/amp.html#transition-guide-for-old-api-users
I plan to get this example working in the next two days, and you can compare with your use case.

**Comment by mcarilli** [comment]
Deduplicating to https://github.com/NVIDIA/apex/issues/179.

**Comment by hukkelas** [comment]
Thank you for the answer. I actually took a look at the new API refactor the day after I posted this and tried to implement a GAN training with both opt level O1 and O2. 

I did not initialize the running average generator with AMP, but I initialized both the generator/discriminator used for the training steps. 

From the testing I've been able to done, I've seen similar performance for a progressive growing GAN in terms of Freichet Inception Distance, while achieving about a ~30% improvement in convergence speed on a V100-32GB
